Document/0x05e-Testing-Cryptography.md

```diff
@@ -1,16 +1,70 @@
 ## Android Cryptographic APIs
 
+### Overview
+
 In the chapter "[Cryptography for Mobile Apps](0x04g-Testing-Cryptography.md)", we introduced general cryptography best practices and described typical flaws that can occur when cryptography is used incorrectly in mobile apps. In this chapter, we'll go into more detail on Android's cryptography APIs. We'll show how to identify uses of those APIs in the source code and how to interpret the configuration. When reviewing code, make sure to compare the cryptographic parameters used with the current best practices linked from this guide.
 
-### Testing the Configuration of Cryptographic Standard Algorithms (MSTG-CRYPTO-2, MSTG-CRYPTO-3 and MSTG-CRYPTO-4)
+We can identify key components of cryptography system in Android:
 
-#### Overview
+- [Provider](0x05e-Testing-Cryptography.md#provider)
+- KeyStore - see the section [KeyStore](0x05d-Testing-Data-Storage.md#keystore) in the chapter "Testing Data Storage"
+- KeyChain - see the section [KeyChain](0x05d-Testing-Data-Storage.md#keychain) in the chapter "Testing Data Storage"
 
 Android cryptography APIs are based on the Java Cryptography Architecture (JCA). JCA separates the interfaces and implementation, making it possible to include several [security providers](https://developer.android.com/reference/java/security/Provider.html "Android Security Providers") that can implement sets of cryptographic algorithms. Most of the JCA interfaces and classes are defined in the `java.security.*` and `javax.crypto.*` packages. In addition, there are Android specific packages `android.security.*` and `android.security.keystore.*`.
 
+KeyStore and KeyChain provides APIs for storing and using keys (behind the scene, KeyChain API uses KeyStore system). These systems allow to administer the full lifecycle of the cryptographic keys. Requirements and guidance for implementation of cryptographic key management can be found in [Key Management Cheat Sheet](https://owasp.org/www-project-cheat-sheets/cheatsheets/Key_Management_Cheat_Sheet.html "Key Management Cheat Sheet"). We can identify following phases:
+
+- generating of a key
+- using of a key
+- storing of a key
+- archiving of a key
+- deleting of a key
+
+> Please note that storing of a key is analyzed in the chapter "[Testing Data Storage](0x05d-Testing-Data-Storage.md)".
+
+These phases are managed by Keystore/KeyChain system. However how the system works depends on how application developer used it. Then, you should focus on functions which are used by application developer. You should identify and verify following functions:
```

Comment:
let's add Key-rotation here as well: after all every key has a given lifetime given the security put into securing it and the need for attackers to grab it I(e.g. risk of the data being leaked)
.

Reply:
Good point. That topic need `more love` also (at least one section level 4).

Reply:
Added to list of functionalities and added to #1689